types/query.d.ts

```diff
@@ -9,18 +9,18 @@ declare module 'mongoose' {
   type QueryTypeCasting<T> = T extends string
     ? StringQueryTypeCasting
     : T extends Types.ObjectId
-      ? ObjectIdQueryTypeCasting
-      : T extends Types.UUID
-        ? UUIDQueryTypeCasting
-        : T extends Buffer
-          ? BufferQueryCasting
-          : T extends NativeDate
-            ? DateQueryTypeCasting
-            : T;
+    ? ObjectIdQueryTypeCasting
+    : T extends Types.UUID
+    ? UUIDQueryTypeCasting
+    : T extends Buffer
+    ? BufferQueryCasting
+    : T extends NativeDate
+    ? DateQueryTypeCasting
+    : T;
 
   export type ApplyBasicQueryCasting<T> = QueryTypeCasting<T> | QueryTypeCasting<T[]> | (T extends (infer U)[] ? QueryTypeCasting<U> : T) | null;
 
-  type _QueryFilter<T> = ({ [P in keyof T]?: mongodb.Condition<ApplyBasicQueryCasting<T[P]>>; } & mongodb.RootFilterOperators<{ [P in keyof T]?: ApplyBasicQueryCasting<T[P]>; }>);
+  type _QueryFilter<T> = ({ [P in keyof T]?: mongodb.Condition<ApplyBasicQueryCasting<T[P]>>; } & mongodb.RootFilterOperators<{ [P in keyof T]?: ApplyBasicQueryCasting<T[P]>; }> & Record<string, any>);
```

Comment:
This is a bad approach as it allows arbitrary access into what should be a strongly typed object. It feels like trying to solve the issue of a locked door by taking a sledgehammer to the adjacent wall. Sure it works, but there must be a better way.